Cython/Compiler/Symtab.py

```diff
@@ -2589,12 +2603,9 @@ def maybe_ref(arg):
                 if base_entry and not base_entry.type.nogil:
                     error(pos, "Constructor cannot be called without GIL unless all base constructors can also be called without GIL")
                     error(base_entry.pos, "Base constructor defined here.")
-        prev_entry = self.lookup_here(name)
         entry = self.declare_var(name, type, pos,
                                  defining=defining,
                                  cname=cname, visibility=visibility)
-        if prev_entry and not defining:
-            entry.overloaded_alternatives = prev_entry.all_alternatives()
```

Comment:
This definitely isn't replaced. So for the top-level `testclass` you don't set up `overloaded_alternatives` for `a`. Instead you just replace `a()` with `a(int)`

Reply:
Hi, sorry for the late reply. This is done in `Scope`'s `declare` method. The top-level class works as expected. I can call `a` with nothing or an `int` on `testclass`. I can even add more overload and it works.. But I cannot inherit and then add overload. 

@da-woods Is there any way I can split this PR into my original problem, and then the secondary problem of the inherited class's overload?

So my problem is only the fact that the overloads of an external class got lost when inherited. So basically the fact that `testclass2` would not be able to call `a()` and `a(int)`. But `testclass2` would not add an overload.

Reply:
Hi - sorry if this is a bit vague - I'm struggling to find time for a detailed look at the moment.

On splitting the PR:
* I think it's probably fine to split it into two parts if the smaller PR solves one problem and doesn't actively break anything else.
* My suspicion is that there should be one approach that solves both. However I could be wrong. it's easy for me to say that when I haven't actually tried it.